source/simulator/helper_functions.cc

```diff
@@ -1624,12 +1652,16 @@ namespace aspect
     for (const auto &cell : dof_handler.active_cell_iterators())
       if (cell->is_locally_owned())
         {
-          fe_values_C.reinit (cell);
           cell->get_dof_indices (local_dof_indices);
```

Comment:
can you just push this line down to immediately above the first block that uses `local_dof_indices`?